smallab/dashboard/dashboard.py

```diff
@@ -3,29 +3,50 @@
 import math
 import multiprocessing as mp
 import time
+import json
 
 import numpy as np
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import OneHotEncoder
+from os.path import join, exists
+from pathlib import Path
 
 from smallab.dashboard.dashboard_events import (BeginEvent, CompleteEvent, ProgressEvent, LogEvent,
-                                                StartExperimentEvent,
+                                                StartExperimentEvent, RegistrationCompleteEvent,
                                                 RegisterEvent, FailedEvent)
+from smallab.file_locations import get_save_directory
 
 # This is a globally accessible queue which stores the events for the dashboard
 
+encoders = dict()  # feature name : dict("encoder" : OneHotEncoder, "values" : list of values)
```

Comment:
these should be members of time estimator not globals

Reply:
Oh okay I'll fix that